src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -509,14 +701,20 @@ def _get_app_env_and_group(
     if not resource_group.name and not resource_group.exists:
         matched_apps = [c for c in list_containerapp(cmd) if c["name"].lower() == name.lower()]
         if env.name:
-            matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["managedEnvironmentId"])["name"].lower() == env.name.lower()]
+            matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["environmentId"])["name"].lower() == env.name.lower()]
```

Comment:
should we compare the resource id of environment directly?

Reply:
for command:
-n {app_name} --environment {name}
could not get resourceId to filter